README.md
# Nagios Notify

The [nagios-notify] package is a template-based notify script for Nagios.

The templates can be easily edited in a text editor. You can change templates
without restarting Nagios. With advanced templates, you can send rich text
(even images) over jabber if you use nagios-notify-jabber. It depends only on
basic Unix utilities.

[nagios-notify]: https://github.com/glensc/nagios-notify

You should use this package because:

- the templates are easily edited in a text editor
- you won't be worried if the command definition contains shell syntax errors
  (which Nagios happily discards without any trace in logs :/)
- you can change templates without restarting Nagios
- with advanced templates, you can send rich-text (even images!) over jabber if
  you use nagios-notify-jabber
- minimal dependency (just coreutils and awk that you most likely already have
  installed)


ChangeLog
2012-12-14 11:57 +0000 [r12607]  Jacek Konieczny <jajcus/at/pld-linux.org>

	* Makefile: Version: 0.14

2012-12-14 11:52 +0000 [r12606]  Jacek Konieczny <jajcus/at/pld-linux.org>

	* nagios-notify: $(unescape:...) function added to un-escape
	  newlines in $LONGSERVICEOUTPUT$

2012-12-14 11:10 +0000 [r12605]  Jacek Konieczny <jajcus/at/pld-linux.org>

	* nagios-notify: Do not replace '&' with $VAR$ for vars containing
	  '&'

2010-01-15 14:18 +0000 [r11114]  Elan Ruusamäe <glen/at/pld-linux.org>

	* templates/notify-service-by-sms.tmpl: - for acknowledgement type
	  show ack author and message

2009-07-28 10:14 +0000 [r10428-10429]  Elan Ruusamäe <glen/at/pld-linux.org>

	* Makefile: - version 0.13: nagios 3.1 compat

	* nagios-notify: - nagios 3.1 compat (skip last_version,
	  new_version) matching

2009-02-22 11:39 +0000 [r10141]  Arkadiusz Miśkiewicz <arekm/at/pld-linux.org>

	* templates/notify-host-by-sms.tmpl,
	  templates/notify-service-by-sms.tmpl: CONTACTPAGER not
	  CONTACTEMAIL for sms notifications

2009-02-20 11:10 +0000 [r10134]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - easy code to save env for debugging

2008-11-04 22:57 +0000 [r9966]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - allow comparing empty string

2008-11-03 23:13 +0000 [r9949-9953]  Elan Ruusamäe <glen/at/pld-linux.org>

	* Makefile: - 0.12 - NOTE: templates and sample config has been
	  renamed to follow nagios 3.0 naming schema

	* Makefile: - add ChangeLog to package

	* changelog.sh (added): - add from rc-scrits trunk

	* templates/notify-service-by-jabber.tmpl (added),
	  templates/notify-by-jabber.tmpl (removed): - one more rename

	* templates/notify-by-email-conditions.tmpl (removed),
	  templates/notify-host-by-jabber.tmpl (added),
	  templates/notify-service-by-jabber-embedimage.tmpl (added),
	  templates/notify-service-by-sms.tmpl (added),
	  templates/host-notify-by-jabber.tmpl (removed),
	  templates/notify-host-by-email.tmpl (added),
	  templates/host-notify-by-eggdrop.tmpl (removed),
	  templates/host-notify-by-email.tmpl (removed),
	  templates/notify-by-jabber-embedimage.tmpl (removed),
	  templates/notify-by-sms.tmpl (removed),
	  templates/notify-host-by-eggdrop.tmpl (added),
	  templates/notify-service-by-jabber-richtext.tmpl (added),
	  templates/notify-host-by-sms.tmpl (added),
	  templates/host-notify-by-sms.tmpl (removed),
	  templates/notify-by-jabber-richtext.tmpl (removed),
	  templates/notify-service-by-eggdrop.tmpl (added),
	  templates/notify-service-by-email.tmpl (added),
	  templates/notify-by-email.tmpl (removed),
	  templates/notify-by-eggdrop.tmpl (removed),
	  templates/notify-service-by-email-conditions.tmpl (added),
	  nagios-notify.cfg: - sanitize naming of host/templates:
	  s/host-notify-by-/notify-host-by-/
	  s/notify-by-/notify-service-by-/

2008-11-03 22:06 +0000 [r9931]  Elan Ruusamäe <glen/at/pld-linux.org>

	* templates/notify-by-jabber-richtext.tmpl: - add [log] link for
	  event history

2008-10-24 09:39 +0000 [r9916]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - allow matching multiple $(func:args...) on same
	  line

2008-10-12 11:16 +0000 [r9906]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - add $(math:expression)

2008-10-11 20:32 +0000 [r9904-9905]  Elan Ruusamäe <glen/at/pld-linux.org>

	* templates/notify-by-email-conditions.tmpl (added): - service
	  template with conditions. based on one from this tarball:
	  http://sweetdeliah.com/nagios/notify/nagios_notify-0.4.2.tar.gz

	* nagios-notify: - add support for conditional templates

2008-10-11 19:39 +0000 [r9903]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - local vars for functions

2008-10-11 17:13 +0000 [r9902]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - implement $(urlencode:data)

2008-10-11 15:00 +0000 [r9899-9900]  Elan Ruusamäe <glen/at/pld-linux.org>

	* Makefile: - version 0.11

	* nagios-notify: - add $(encode_mime_header:CHARSET,DATA)

2008-10-11 14:55 +0000 [r9897-9898]  Elan Ruusamäe <glen/at/pld-linux.org>

	* templates/notify-by-email.tmpl,
	  templates/host-notify-by-email.tmpl: - encode To header

	* templates/notify-by-email.tmpl,
	  templates/host-notify-by-email.tmpl: - unify email headers

2008-10-11 09:57 +0000 [r9896]  Elan Ruusamäe <glen/at/pld-linux.org>

	* templates/host-notify-by-email.tmpl,
	  templates/notify-by-email.tmpl: - add charset to email templates

2008-05-30 05:49 +0000 [r9782]  Elan Ruusamäe <glen/at/pld-linux.org>

	* INSTALL (added): - plain install instructions

2008-04-08 18:08 +0000 [r9702]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify.cfg: - vim filetype=nagios

2008-04-08 11:22 +0000 [r9698-9700]  Elan Ruusamäe <glen/at/pld-linux.org>

	* Makefile: - version 0.10

	* Makefile: - move tool to /usr/lib/nagios

	* nagios-notify.cfg: - move nagios notify outside $PATH as reqires
	  nagios internal vars to operate properly

2008-04-08 11:10 +0000 [r9696-9697]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - UTF8

	* nagios-notify: - configure-like variable test check - update
	  requires - expand Id keyword

2008-04-08 10:01 +0000 [r9695]  Arkadiusz Miśkiewicz <arekm/at/pld-linux.org>

	* nagios-notify: Use default /var/lib/nagios/status.dat when
	  NAGIOS_STATUSDATAFILE is set but empty. Raise error when variable
	  is not set at all. (needed for some nagios3 where
	  NAGIOS_STATUSDATAFILE is set but empty)

2008-01-22 21:57 +0000 [r9196]  Elan Ruusamäe <glen/at/pld-linux.org>

	* nagios-notify: - spelling

2008-01-22 15:01 +0000 [r9192-9194]  Elan Ruusamäe <glen/at/pld-linux.org>

	* templates/notify-by-jabber-embedimage.tmpl: - template with embed
	  image links

	* templates/notify-by-jabber-embedimage.tmpl (added): - cp
	  notify-by-jabber-richtext.tmpl notify-by-jabber-embedimage.tmpl

	* nagios-notify: - convert to awk only

2008-01-22 14:55 +0000 [r9190-9191]  Elan Ruusamäe <glen/at/pld-linux.org>


env.sh
export NAGIOS_ADMINEMAIL='nagios'
export NAGIOS_ADMINPAGER='pagenagios'
export NAGIOS_ARG10=''
export NAGIOS_ARG11=''
export NAGIOS_ARG12=''
export NAGIOS_ARG13=''
export NAGIOS_ARG14=''
export NAGIOS_ARG15=''
export NAGIOS_ARG16=''
export NAGIOS_ARG17=''
export NAGIOS_ARG18=''
export NAGIOS_ARG19=''
export NAGIOS_ARG1=''
export NAGIOS_ARG20=''
export NAGIOS_ARG21=''
export NAGIOS_ARG22=''
export NAGIOS_ARG23=''
export NAGIOS_ARG24=''
export NAGIOS_ARG25=''
export NAGIOS_ARG26=''
export NAGIOS_ARG27=''
export NAGIOS_ARG28=''
export NAGIOS_ARG29=''
export NAGIOS_ARG2=''
export NAGIOS_ARG30=''
export NAGIOS_ARG31=''
export NAGIOS_ARG32=''
export NAGIOS_ARG3=''
export NAGIOS_ARG4=''
export NAGIOS_ARG5=''
export NAGIOS_ARG6=''
export NAGIOS_ARG7=''
export NAGIOS_ARG8=''
export NAGIOS_ARG9=''
export NAGIOS_COMMANDFILE='/var/lib/nagios/rw/nagios.cmd'
export NAGIOS_COMMENTDATAFILE='/var/lib/nagios/comments.dat'
export NAGIOS_CONTACTALIAS='Nagios Admin'
export NAGIOS_CONTACTEMAIL='root@localhost'
export NAGIOS_CONTACTNAME='nagios-admin'
export NAGIOS_CONTACTPAGER='root@localhost'
export NAGIOS_DATE='2006-08-03'
export NAGIOS_DOWNTIMEDATAFILE='/var/lib/nagios/downtime.dat'
export NAGIOS_HOSTACKAUTHOR='glen'
export NAGIOS_HOSTACKCOMMENT='asdfdsf'
export NAGIOS_HOSTACTIONURL=''
export NAGIOS_HOSTADDRESS='127.0.0.1'
export NAGIOS_HOSTALIAS='Linux Server'
export NAGIOS_HOSTATTEMPT='1'
export NAGIOS_HOSTCHECKCOMMAND='check-host-alive'
export NAGIOS_HOSTCHECKTYPE='ACTIVE'
export NAGIOS_HOSTDOWNTIME='0'
export NAGIOS_HOSTDURATION='2d 18h 53m 45s'
export NAGIOS_HOSTDURATIONSEC='240825'
export NAGIOS_HOSTEXECUTIONTIME='0.003'
export NAGIOS_HOSTGROUPALIAS='Linux Servers'
export NAGIOS_HOSTGROUPNAME='test'
export NAGIOS_HOSTLATENCY='0.000'
export NAGIOS_HOSTNAME='localhost'
export NAGIOS_HOSTNOTES='PLD Linux server'
export NAGIOS_HOSTNOTESURL=''
export NAGIOS_HOSTOUTPUT='(No output!)'
export NAGIOS_HOSTPERCENTCHANGE='0.00'
export NAGIOS_HOSTPERFDATA=''
export NAGIOS_HOSTPERFDATAFILE='/var/lib/nagios/host-perfdata'
export NAGIOS_HOSTSTATE='DOWN'
export NAGIOS_HOSTSTATEID='1'
export NAGIOS_HOSTSTATETYPE='HARD'
export NAGIOS_LASTHOSTCHECK='1154602028'
export NAGIOS_LASTHOSTDOWN='1154602028'
export NAGIOS_LASTHOSTSTATECHANGE='1154361205'
export NAGIOS_LASTHOSTUNREACHABLE='0'
export NAGIOS_LASTHOSTUP='1154361025'
export NAGIOS_LASTSERVICECHECK=''
export NAGIOS_LASTSERVICECRITICAL=''
export NAGIOS_LASTSERVICEOK=''
export NAGIOS_LASTSERVICESTATECHANGE=''
export NAGIOS_LASTSERVICEUNKNOWN=''
export NAGIOS_LASTSERVICEWARNING=''
export NAGIOS_LOGFILE='/var/log/nagios/nagios.log'
export NAGIOS_LONGDATETIME='Thu Aug 3 13:47:10 EEST 2006'
export NAGIOS_MAINCONFIGFILE='/etc/nagios/nagios.cfg'
export NAGIOS_NOTIFICATIONNUMBER='34'
export NAGIOS_NOTIFICATIONTYPE='ACKNOWLEDGEMENT'
export NAGIOS_OBJECTCACHEFILE='/var/lib/nagios/objects.cache'
export NAGIOS_PROCESSSTARTTIME='1154601778'
export NAGIOS_RESOURCEFILE='/etc/nagios/resource.cfg'
export NAGIOS_RETENTIONDATAFILE='/var/lib/nagios/retention.dat'
export NAGIOS_SERVICEACKAUTHOR=''
export NAGIOS_SERVICEACKCOMMENT=''
export NAGIOS_SERVICEACTIONURL=''
export NAGIOS_SERVICEATTEMPT=''
export NAGIOS_SERVICECHECKCOMMAND=''
export NAGIOS_SERVICECHECKTYPE=''
export NAGIOS_SERVICEDESC='service#desc'
export NAGIOS_SERVICEDOWNTIME=''
export NAGIOS_SERVICEDURATION=''
export NAGIOS_SERVICEDURATIONSEC=''
export NAGIOS_SERVICEEXECUTIONTIME=''
export NAGIOS_SERVICEGROUPALIAS=''
export NAGIOS_SERVICEGROUPNAME=''
export NAGIOS_SERVICELATENCY=''
export NAGIOS_SERVICENOTES=''
export NAGIOS_SERVICENOTESURL=''
export NAGIOS_SERVICEOUTPUT=''
export NAGIOS_SERVICEPERCENTCHANGE=''
export NAGIOS_SERVICEPERFDATA=''
export NAGIOS_SERVICEPERFDATAFILE='/var/lib/nagios/service-perfdata'
export NAGIOS_SERVICESTATE=''
export NAGIOS_SERVICESTATEID=''
export NAGIOS_SERVICESTATETYPE=''
export NAGIOS_SHORTDATETIME='2006-08-03 13:47:10'
export NAGIOS_STATUSDATAFILE='/var/lib/nagios/status.dat'
export NAGIOS_TEMPFILE='/var/lib/nagios/nagios.tmp'
export NAGIOS_TIME='13:47:10'
export NAGIOS_TIMET='1154602030'
export NAGIOS_TOTALHOSTPROBLEMS='1'
export NAGIOS_TOTALHOSTPROBLEMSUNHANDLED='1'
export NAGIOS_TOTALHOSTSDOWN='1'
export NAGIOS_TOTALHOSTSDOWNUNHANDLED='1'
export NAGIOS_TOTALHOSTSUNREACHABLE='0'
export NAGIOS_TOTALHOSTSUNREACHABLEUNHANDLED='0'
export NAGIOS_TOTALHOSTSUP='0'
export NAGIOS_TOTALSERVICEPROBLEMS='2'
export NAGIOS_TOTALSERVICEPROBLEMSUNHANDLED='0'
export NAGIOS_TOTALSERVICESCRITICAL='1'
export NAGIOS_TOTALSERVICESCRITICALUNHANDLED='0'
export NAGIOS_TOTALSERVICESOK='3'
export NAGIOS_TOTALSERVICESUNKNOWN='1'
export NAGIOS_TOTALSERVICESUNKNOWNUNHANDLED='0'
export NAGIOS_TOTALSERVICESWARNING='0'
export NAGIOS_TOTALSERVICESWARNINGUNHANDLED='0'
